src/treelearner/feature_histogram.hpp

```diff
@@ -80,44 +85,61 @@ class FeatureHistogram {
     }
   }
 
-  void FindBestThreshold(double sum_gradient, double sum_hessian, data_size_t num_data, double min_constraint, double max_constraint,
-                         SplitInfo* output) {
+  void FindBestThreshold(double sum_gradient, double sum_hessian,
+                         data_size_t num_data, SplitInfo *output,
+                         SplittingConstraints &constraints) {
```

Comment:
use pointer type for `constraints`, so that we could just run it only when it is not `nullptr`.